es4x/src/main/java/io/reactiverse/es4x/MJSVerticleFactory.java

```diff
@@ -23,15 +23,15 @@
 import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
 
-public final class MJSVerticleFactory extends ESVerticleFactory {
```

Comment:
I think these factories should by impl private. You can add new ones using the vertx service loader and if the abstract class makes the create verticle open then you would have full access to the code.

Reply:
You're right but was able to reuse MJS/JS VerticleFactory with only init method  overriden. So if someone wanted to use engine to put some members it will be simple to just extends those factories and hook into the init method.